controller/array_action/array_mediator_ds8k.py

```diff
@@ -353,19 +356,16 @@ def get_iscsi_targets_by_iqn(self):
         return {}
 
     def get_array_fc_wwns(self, host_name=None):
-        logger.debug("Getting the connected fc port wwpns from array")
+        logger.debug("Getting the connected fc port wwpns for host {} from array".format(host_name))
 
-        # remove this line when pyds8k support get_ioports_by_host
-        host_name = None
         try:
-            if host_name:
-                fc_ports = self.client.get_ioports_by_host(host_name)
-            else:
-                fc_ports = self.client.get_fcports()
-
-            wwpns = [p.wwpn for p in fc_ports if p.state == IOPORT_STATUS_ONLINE]
+            host = self.client.get_host(host_name)
+            wwpns = [port[LOGIN_PORT_WWPN] for port in host.login_ports if
+                     port[LOGIN_PORT_STATE] == LOGIN_PORT_STATE_ONLINE]
             logger.debug("Found wwpns: {}".format(wwpns))
             return wwpns
+        except exceptions.NotFound:
```

Comment:
change `try` content to include only the `host = self.client.get_host(host_name)` line

Reply:
I prefer this way
host is defined inside try
other methods are done similarly. See ds8k mediator method get_volume_mappings for example

Reply:
`get_volume_mappings` has a loop in the try, so at least it avoids multiplying the try overhead.

if you don't want a big `try/except` block in the middle of the function, make a separate function for it. as written in the Clean Code book:
"if the keyword 'try' exists in a function, it should be the very first word in the function and that there should be nothing after the catch/finally blocks."

besides that, I don't see another reason to `try/except` irrelevant code lines.

Reply:
" there should be nothing after the catch/finally blocks."
so where to put the rest of the  code? (i don't want separate func. this is overkill)

Reply:
see create_volume in xiv mediator. he same - command after xcli command.

Reply:
I don't think it's an overkill, it's just another small function. small functions are good.
and I don't agree with how other functions did it. they are big and ugly.

Reply:
if others work this way i won'g change the style